### PR TITLE
Add MySQL parse support for `TABLE Statement`

### PR DESCRIPTION
**Pull request by totalo**
For #14110.

Changes proposed in this pull request:
- add MySQL parse support for `TABLE Statement`
- add test case

### CODE REVIEW

**Review by tuichenchuxin — APPROVED** [review]

**Comment by tuichenchuxin** [comment]
@totalo This branch has conflicts that must be resolved.

**Comment by totalo** [comment]
> @totalo This branch has conflicts that must be resolved.

All right! done.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/dml/table.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<sql-cases>
+    <sql-case id="table_with_order_limit_offset" value="TABLE t_order ORDER BY order_id LIMIT 1 OFFSET 2" db-types="MySQL"/>
+</sql-cases>
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/dml/table.xml:21]: @totalo Please add a new line here.

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dml/table.xml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+<sql-parser-test-cases>
+    <table sql-case-id="table_with_order_limit_offset">
+        <simple-table name="t_order" start-index="6" stop-index="12" />
+        <column name="order_id" start-index="23" stop-index="30"/>
+        <limit literal-start-index="32" literal-stop-index="47">
+            <offset value="2" literal-start-index="47" literal-stop-index="47" />
+            <row-count value="1" literal-start-index="38" literal-stop-index="38" />
+        </limit>
+    </table>
+</sql-parser-test-cases>
```
Comment (strongduanmu) [inline shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dml/table.xml:28]: @totalo Please add a new line here.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dml/MySQLTableStatement.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.apache.shardingsphere.sql.parser.sql.common.statement.dml.TableStatement;
+import org.apache.shardingsphere.sql.parser.sql.dialect.statement.mysql.MySQLStatement;
+
+/**
+ * MySQL table statement.
+ */
+@Getter
+@Setter
+@ToString
+public class MySQLTableStatement extends TableStatement implements MySQLStatement {
```
Comment (strongduanmu) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dml/MySQLTableStatement.java:35]: Please add final for MySQLTableStatement.

**Review by strongduanmu — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/dml/table.xml (hunk 1 above)
Comment (totalo) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/sql/supported/dml/table.xml]: done.thx

File: shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dml/table.xml (hunk 2 above)
Comment (totalo) [reply shardingsphere-test/shardingsphere-parser-test/src/main/resources/case/dml/table.xml]: done.thx

File: shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dml/MySQLTableStatement.java (hunk 3 above)
Comment (totalo) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-statement/src/main/java/org/apache/shardingsphere/sql/parser/sql/dialect/statement/mysql/dml/MySQLTableStatement.java]: done.thx

**Review by strongduanmu — APPROVED** [review]

**Pull request merged**